enhancements/olm/safe-cluster-upgrades.md

```diff
@@ -0,0 +1,263 @@
+---
+title: Safe OpenShift Upgrades with Operators 
+authors:
+  - "@awgreene"
+reviewers:
+  - TBD
+  - "@kevinrizza"
+approvers:
+  - "@ecordell"
+  - "@spadgett"
+creation-date: 2021-01-11
+last-updated: 2021-01-19
+status: provisional
+see-also:
+  -  N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# Safe OpenShift Upgrades with Operators
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Definitions
+
+**OLM Managed Operators**: Operators that are installed, managed, and upgraded by the [Operator Lifecycle Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) project.
+
+**Minor OpenShift Upgrade**: When the OpenShift Cluster is upgraded to the next minor version, for example the upgrade from version 4.6 to 4.7. See [Semantic Versioning](https://semver.org/) for more details.
+
+## Summary
+
+OpenShift Cluster Admins introduce new services to their clusters in the form of OLM Managed Operators. When performing a cluster upgrade, there is no guarantee that these operators will continue to run on the upgraded cluster.
+
+OLM must make an effort to prevent cluster upgrades when it is possible to determine that one or more installed operators will not run on the next version of OpenShift.
+
+## Motivation
+
+The primary purpose of this enhancement is to define how OpenShift can protect existing services introduced by OLM Managed Operators when performing a Minor OpenShift upgrade.
+
+### Goals
+
+- Prevent Minor OpenShift Upgrades  when an installed operator declares that it does not support the next ocp minor version
+- Warn the Cluster Admin when an installed operator does not explicitly declare support for the next ocp minor version
+
+### Non-Goals
+
+- Ensure that services will not be disrupted during a cluster upgrade.
+- Guarantee that all installed operator will continue to run after a cluster upgrade.
+- Prevent Patch OpenShift Upgrades.
+
+## Proposal
+
+The [ClusterVersionOperator](https://github.com/openshift/cluster-version-operator) provides OLM with the means to prevent Minor Semantic Version cluster upgrades by setting the  `operator-lifecycle-manager` [ClusterOperator's](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusteroperator.md) [upgradeable condition](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusteroperator.md#what-should-an-operator-report-with-clusteroperator-custom-resource) to `False`.
```

Comment:
nit: `master` links are unreliable, because things might be changed or moved.  Probably best to pin to a specific commit, and also probably better to link the official API instead of CVO's dev docs.  Like [this][1].

[1]: https://github.com/openshift/api/blob/5935a5beec4bb8e1e81dd0fe9ebc2af36b9a09ae/config/v1/types_cluster_operator.go#L171-L175